File: 1544. Make The String Great/solution.py
class Solution(object):
  def makeGood(self, s):
    """
    :type s: str
    :rtype: str
    """
    myStack = []
    for letter in s:
      if len(myStack) == 0:
        myStack.append(letter)
      elif myStack[-1].isupper() and myStack[-1].lower() == letter:
        myStack.pop()
      elif myStack[-1].islower() and myStack[-1].upper() == letter:
        myStack.pop()
      else:
        myStack.append(letter)
    
    return ''.join(myStack)

File: 1544. Make The String Great/solution.cpp
class Solution {
public:
  string makeGood(string s) {
    vector<char> myStack;
    for (int index = 0; index < s.size(); index++){
      char letter = s[index];
      if (myStack.size() == 0){
        myStack.push_back(letter);      
      }
      else if (isupper(myStack[myStack.size() - 1]) && tolower(myStack[myStack.size() - 1]) == letter){
        myStack.pop_back();      
      }
      else if (islower(myStack[myStack.size() - 1]) && toupper(myStack[myStack.size() - 1]) == letter){
        myStack.pop_back();
      }
      else{
        myStack.push_back(letter);
      }
    }

    string result;
    for (int index = 0; index < myStack.size(); index++){
      result += myStack[index];
    }
    return result;
  }
};